app/src/main/java/ch/epfl/qedit/view/util/EditTextDialog.java

```diff
@@ -0,0 +1,159 @@
+    private TextFilter textFilter = NO_FILTER;
+    private String message;
+    private EditText editText;
+    private AlertDialog dialog;
+
+    private EditTextDialog() {}
+
+    /**
+     * The edit text dialogs should be created through this method, because Fragments cannot have
+     * constructors that take arguments.
+     *
+     * @param message the message to display above the edit text view
+     * @param listener the listener that gets notified once the text was entered
+     * @return a new dialog
+     */
+    public static EditTextDialog create(String message, SubmissionListener listener) {
+        EditTextDialog dialog = new EditTextDialog();
+
+        Bundle args = new Bundle();
+        args.putString("message", message);
+        args.putSerializable("listener", listener);
+
+        dialog.setArguments(args);
+        return dialog;
+    }
+
+    /**
+     * This method allows to validate the text that the user enters in the dialog. The text filter
+     * can return an error to display, or null to indicate that the text is valid. The constant
+     * NO_FILTER can be used to validate everything.
+     *
+     * @param textFilter the text filter, non null.
+     */
+    public void setTextFilter(TextFilter textFilter) {
+        this.textFilter = Objects.requireNonNull(textFilter);
+    }
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        editText = new EditText(context);
+        editText.setInputType(InputType.TYPE_CLASS_TEXT);
+        editText.addTextChangedListener(
+                new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(
+                            CharSequence s, int start, int count, int after) {}
+
+                    @Override
+                    public void onTextChanged(CharSequence s, int start, int before, int count) {
+                        String error = textFilter.isAllowed(s.toString());
+                        Button positive = dialog.getButton(DialogInterface.BUTTON_POSITIVE);
+
+                        if (error != null) {
+                            editText.setError(error);
+                            positive.setEnabled(false);
+                        } else {
+                            positive.setEnabled(true);
+                        }
+                    }
+
+                    @Override
+                    public void afterTextChanged(Editable s) {}
+                });
+
+        // This is for testing, so that the text can be retrieved by id
+        editText.setId(R.id.quiz_name_text);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        Bundle args = getArguments();
+        message = args.getString("message");
+        listener = (SubmissionListener) args.getSerializable("listener");
+
+        dialog =
+                new AlertDialog.Builder(getActivity())
```

Comment:
RequireActivity() would possibly be safer.

Reply:
Yes, I will also change that, that's a mistake from me -_-